### ISSUE TITLE
Environment not recognized?

### ISSUE THREAD

**Issue by gabegrand** [ppaquette/gym-super-mario#1]
Hello, I can't seem to get gym_pull working properly with SMB. It seems like there is an issue with the install location. Any ideas? I'm on Mac OSX Yosemite with a Python 2.7 anaconda distribution.

```
>>> import gym, gym_pull
>>> gym_pull.pull('github.com/ppaquette/gym-super-mario')
[2016-10-23 19:09:50,870] Installing pip package from "https://github.com/ppaquette/gym-super-mario.git"
Collecting git+https://github.com/ppaquette/gym-super-mario.git
  Cloning https://github.com/ppaquette/gym-super-mario.git to /var/folders/p5/m1z33npj37z93fws69vxykv80000gn/T/pip-MThd2b-build
Requirement already up-to-date: gym>=0.2.3 in /Users/ggrand/Desktop/gym (from ppaquette-gym-super-mario==0.0.1)
Requirement already up-to-date: numpy>=1.10.4 in /anaconda/lib/python2.7/site-packages (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Requirement already up-to-date: requests>=2.0 in /anaconda/lib/python2.7/site-packages (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Requirement already up-to-date: six in /anaconda/lib/python2.7/site-packages/six-1.10.0-py2.7.egg (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Requirement already up-to-date: pyglet>=1.2.0 in /anaconda/lib/python2.7/site-packages (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Installing collected packages: ppaquette-gym-super-mario
  Found existing installation: ppaquette-gym-super-mario 0.0.1
    Uninstalling ppaquette-gym-super-mario-0.0.1:
      Successfully uninstalled ppaquette-gym-super-mario-0.0.1
  Running setup.py install for ppaquette-gym-super-mario ... done
Successfully installed ppaquette-gym-super-mario-0.0.1
[2016-10-23 19:09:53,381] The user environments for "github.com/ppaquette/gym-super-mario" are already up-to-date (no new version detected).
>>> env = gym.make('ppaquette/SuperMarioBros-1-1-v0')
[2016-10-23 19:10:20,645] Making new env: ppaquette/SuperMarioBros-1-1-v0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "gym/envs/registration.py", line 126, in make
    return registry.make(id)
  File "gym/envs/registration.py", line 90, in make
    spec = self.spec(id)
  File "gym/envs/registration.py", line 112, in spec
    raise error.UnregisteredEnv('No registered env with id: {}'.format(id))
gym.error.UnregisteredEnv: No registered env with id: ppaquette/SuperMarioBros-1-1-v0
```

**Comment by ppaquette** [comment]
I fixed a couple of bugs in gym-pull yesterday. Make sure you are using v0.1.5 and copy the output of the following commands:

```
>>> import gym
>>> import gym_pull
>>> [env for env in gym_pull.list() if 'ppaquette' in env]
>>> gym_pull.pull('github.com/ppaquette/gym-super-mario')
>>> [env for env in gym_pull.list() if 'ppaquette' in env]
>>> import ppaquette_gym_super_mario
>>> print(ppaquette_gym_super_mario.__file__)
>>> [env for env in gym_pull.list() if 'ppaquette' in env]
```

and
`cat /Users/ggrand/Desktop/gym/gym/envs/.envs.json`

**Comment by gabegrand** [comment]
Hi Philip, here's the output from running those commands in terminal. By the way, I used to have the gym directory cloned on my Desktop, but then I deleted it and installed gym using pip in an attempt to fix the issue with gym-pull. I noticed that pip is complaining about that, so I figured it'd be worth mentioning.

Thanks in advance for your help.

```
Gabes-MacBook-Pro-2:~ ggrand$ pip show gym-pull
---
Metadata-Version: 2.0
Name: gym-pull
Version: 0.1.5
Summary: Add-on for OpenAI Gym that supports automatic downloading of user environments.
Home-page: https://github.com/ppaquette/gym-pull
Author: Philip Paquette
Author-email: pcpaquette@gmail.com
Installer: pip
License: MIT License
Location: /anaconda/lib/python2.7/site-packages
Requires: gym, six
Classifiers:
Gabes-MacBook-Pro-2:~ ggrand$ python
Python 2.7.11 |Anaconda 2.3.0 (x86_64)| (default, Dec  6 2015, 18:57:58) 
[GCC 4.2.1 (Apple Inc. build 5577)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import gym
>>> import gym_pull
>>> [env for env in gym_pull.list() if 'ppaquette' in env]
[]
>>> gym_pull.pull('github.com/ppaquette/gym-super-mario')
[2016-10-24 16:35:23,739] Installing pip package from "https://github.com/ppaquette/gym-super-mario.git"
Collecting git+https://github.com/ppaquette/gym-super-mario.git
  Cloning https://github.com/ppaquette/gym-super-mario.git to /var/folders/p5/m1z33npj37z93fws69vxykv80000gn/T/pip-wvSgBY-build
Collecting gym>=0.2.3 (from ppaquette-gym-super-mario==0.0.1)
  Downloading gym-0.4.9.tar.gz (141kB)
    100% |████████████████████████████████| 143kB 1.6MB/s 
Requirement already up-to-date: numpy>=1.10.4 in /anaconda/lib/python2.7/site-packages (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Requirement already up-to-date: requests>=2.0 in /anaconda/lib/python2.7/site-packages (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Requirement already up-to-date: six in /anaconda/lib/python2.7/site-packages/six-1.10.0-py2.7.egg (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Requirement already up-to-date: pyglet>=1.2.0 in /anaconda/lib/python2.7/site-packages (from gym>=0.2.3->ppaquette-gym-super-mario==0.0.1)
Building wheels for collected packages: gym
  Running setup.py bdist_wheel for gym ... done
  Stored in directory: /Users/ggrand/Library/Caches/pip/wheels/97/48/bf/ccfb5f1f105edb07930934297e4c0f97ad7ed1cb9fe6db338e
Successfully built gym
Installing collected packages: gym, ppaquette-gym-super-mario
  Found existing installation: gym 0.4.8
Exception:
Traceback (most recent call last):
  File "//anaconda/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "//anaconda/lib/python2.7/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "//anaconda/lib/python2.7/site-packages/pip/req/req_set.py", line 736, in install
    requirement.uninstall(auto_confirm=True)
  File "//anaconda/lib/python2.7/site-packages/pip/req/req_install.py", line 687, in uninstall
    '(at %s)' % (link_pointer, self.name, dist.location)
AssertionError: Egg-link /Users/ggrand/Desktop/gym does not match installed location of gym (at /anaconda/lib/python2.7/site-packages)
>>> [env for env in gym_pull.list() if 'ppaquette' in env]
[]
>>> import ppaquette_gym_super_mario
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "//anaconda/lib/python2.7/site-packages/ppaquette_gym_super_mario/__init__.py", line 4, in <module>
    from .nes_env import NesEnv, MetaNesEnv
  File "//anaconda/lib/python2.7/site-packages/ppaquette_gym_super_mario/nes_env.py", line 20, in <module>
    raise gym.error.DependencyNotInstalled("fceux is required. Try installing with apt-get install fceux.")
gym.error.DependencyNotInstalled: fceux is required. Try installing with apt-get install fceux.
>>> print(ppaquette_gym_super_mario.__file__)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'ppaquette_gym_super_mario' is not defined
>>> [env for env in gym_pull.list() if 'ppaquette' in env]
[]

Gabes-MacBook-Pro-2:~ ggrand$ cat /Users/ggrand/Desktop/gym/gym/envs/.envs.json
cat: /Users/ggrand/Desktop/gym/gym/envs/.envs.json: No such file or directory

```

**Comment by ppaquette** [comment]
There are 2 problems:

1) Conflict between gym packages.

```
AssertionError: Egg-link /Users/ggrand/Desktop/gym does not match installed location of gym (at /anaconda/lib/python2.7/site-packages)
```

2) fceux (NES emulator) is not installed.

```
gym.error.DependencyNotInstalled: fceux is required. Try installing with apt-get install fceux.
```

The gym-pull package is just a wrapper that installs the package `ppaquette_gym_super_mario` by running `pip install git+https://github.com/ppaquette/gym-super-mario/` and then imports it automatically on import by running `import ppaquette_gym_super_mario`.

The ppaquette_gym_super_mario registers its envs on import with gym.

**Comment by gabegrand** [comment]
Hi Philip, I uninstalled and reinstalled both the gym and gym_pull packages using pip. I also installed fceux using brew. I was able to get the emulator working, but only after running the command `import ppaquette_gym_super_mario`. Before that, running the command `[env for env in gym_pull.list() if 'ppaquette' in env]` would return an empty array.

I also have some questions about how to pipe actions into the emulator, but since that's a separate issue, I'll close this issue and open a new one.
